pkg/ottl/expression.go

```diff
@@ -245,6 +246,39 @@ func (g StandardFloatGetter[K]) Get(ctx context.Context, tCtx K) (float64, error
 	}
 }
 
+type FunctionGetter[K any] interface {
+	Get(ctx context.Context, iArgs Arguments) (Expr[K], error)
+}
+
+type StandardFunctionGetter[K any] struct {
+	fCtx FunctionContext
+	fact Factory[K]
+}
+
+func (g StandardFunctionGetter[K]) Get(_ context.Context, iArgs Arguments) (Expr[K], error) {
+	if g.fact == nil {
+		return Expr[K]{}, fmt.Errorf("undefined function")
+	}
+	fArgs := g.fact.CreateDefaultArguments()
+	fArgsVal := reflect.ValueOf(fArgs).Elem()
+	iArgsVal := reflect.ValueOf(iArgs).Elem()
+
+	for i := 0; i < fArgsVal.NumField(); i++ {
+		field := iArgsVal.Field(i)
+		// The Argument passed in could be of any type, function Eval() will handle the type checking
+		fArgsVal.Field(i).Set(field)
+	}
+	fn, err := g.fact.CreateFunction(g.fCtx, fArgs)
+	if err != nil {
+		return Expr[K]{}, fmt.Errorf("couldn't create function: %w", err)
+	}
+	return Expr[K]{exprFunc: fn}, nil
+}
+
+func NewFunctionGetter[K any](fCtx FunctionContext, fact Factory[K]) FunctionGetter[K] {
```

Comment:
I don't think we need this function.  Similar to `StandardStringGetter` with can export `fCtx` and `fact` from `StandardFunctionGetter`.

Reply:
ok, let me try this out.

Reply:
made this change